### ISSUE TITLE
Link.TargetNode is not set

### ISSUE THREAD

**Issue by 314159265meow** [Blazor-Diagrams/Blazor.Diagrams#131]
Hi, 
I noticed that the `TargetNode` property of a Link is not set when the link is created manually.
I tested it with the "Simple" demo page: the Links that are created in the `OnInitialized()` function have a valid `TargetNode` property, when i create a manually the property is `null`. One can still access the target node with the `Link.TargetPort.Parent` property though.
Is that supposed to be that way and do I have to set the TargetNode property manually somehow or is this a bug?

**Comment by zHaytam** [comment]
Hello,

Links that are created between 2 ports shouldn't have the `SourceNode` and `TargetNode` set. These are only set when a link is created between 2 nodes, not ports. That's how I know a link is Portless (between nodes).

The properties are set using one of the two types of constructors:
```cs
public LinkModel(PortModel sourcePort, PortModel? targetPort = null) : base(sourcePort, targetPort) { }
public LinkModel(NodeModel sourceNode, NodeModel? targetNode) : base(sourceNode, targetNode) { }
```

Are you sure you saw the `TargetNode` property set, after creating a link betwene 2 ports?

**Comment by zHaytam** [comment]
Hello,

How do you create the link manually? Do you mean programmatically?
If you create a link between two ports, then the Target/SourceNode properties are set based on the ports' parents, can you tell me how you get the issue please?

**Comment by 314159265meow** [comment]
With manually I mean creating a link by clicking and dragging in the diagram.

To be sure it's not from my code, i tested it with `samples/SharedDemo/Demos/Simple.razor`:

For simplicity I changed the `ToggleVirtualization()` Eventhandler to the following code
```
        protected void ToggleVirtualization()
        {
            foreach (var link in diagram.Links)
                System.Console.WriteLine($"{link.Id}: {link.SourceNode?.Id ?? "#"} - {link.TargetNode?.Id ?? "#"} || {link.SourcePort?.Parent?.Id ?? "#"} - {link.TargetPort?.Parent?.Id ?? "#"}");
        }
```

When I start the demo and click the button, the output is the following:

```
﻿adbb5308-7247-46ff-aed5-3d94ca43a412: 1e3ebcfb-4006-414e-b4e2-a67f57172796 - 748a6a49-bd29-410a-aaa6-20611beb3b5b || 1e3ebcfb-4006-414e-b4e2-a67f57172796 - 748a6a49-bd29-410a-aaa6-20611beb3b5b
305d1080-7c64-449b-b202-5b0d362cf48a: 748a6a49-bd29-410a-aaa6-20611beb3b5b - e27ef584-c3b7-4fdc-bde6-d53a4f2a1abf || 748a6a49-bd29-410a-aaa6-20611beb3b5b - e27ef584-c3b7-4fdc-bde6-d53a4f2a1abf
```

These are the two links that are created in the `OnInitialized()` method. 
When I now create another link by clicking and dragging, the output is the following
```
adbb5308-7247-46ff-aed5-3d94ca43a412: 1e3ebcfb-4006-414e-b4e2-a67f57172796 - 748a6a49-bd29-410a-aaa6-20611beb3b5b || 1e3ebcfb-4006-414e-b4e2-a67f57172796 - 748a6a49-bd29-410a-aaa6-20611beb3b5b
305d1080-7c64-449b-b202-5b0d362cf48a: 748a6a49-bd29-410a-aaa6-20611beb3b5b - e27ef584-c3b7-4fdc-bde6-d53a4f2a1abf || 748a6a49-bd29-410a-aaa6-20611beb3b5b - e27ef584-c3b7-4fdc-bde6-d53a4f2a1abf
15c7ad0e-3476-4e8e-a4b8-3aacec553ff4: 1e3ebcfb-4006-414e-b4e2-a67f57172796 - # || 1e3ebcfb-4006-414e-b4e2-a67f57172796 - e27ef584-c3b7-4fdc-bde6-d53a4f2a1abf
```

The new link has `TargetPort.Parent` set, but `TargetNode` is empty.

**Comment by zHaytam** [comment]
This is now available in 2.1.4!
